msteams-platform/toolkit/TeamsFx-SDK.md

```diff
@@ -402,11 +342,230 @@ The `@microsoft/mgt-teamsfx-provider` package exposes the `TeamsFxProvider` clas
     }
     ```
 
-For more information on sample to initialize the TeamsFx provider, see the [Contacts Exporter sample](https://github.com/OfficeDev/TeamsFx-Samples/tree/dev/hello-world-tab-with-backend).
+    For more information on sample to initialize the TeamsFx provider, see the [Contacts Exporter sample](https://github.com/OfficeDev/TeamsFx-Samples/tree/dev/graph-toolkit-contact-exporter).
 
-</details>
+    </details>
+
+  * User delegate permission in backend (Use OnBehalfOfUserCredential)
+    <details>
+    <summary><b>Use Graph API in bot Application</b></summary>
+
+    1. Initialize and add `TeamsBotSsoPrompt` to dialog set.
+
+    ```ts
+    const { ConversationState, MemoryStorage } = require("botbuilder");
+    const { DialogSet, WaterfallDialog } = require("botbuilder-dialogs");
+    const { TeamsBotSsoPrompt } = require("@microsoft/teamsfx");
+
+    const convoState = new ConversationState(new MemoryStorage());
+    const dialogState = convoState.createProperty("dialogState");
+    const dialogs = new DialogSet(dialogState);
+
+    const teamsfx = new TeamsFx();
+    dialogs.add(
+      new TeamsBotSsoPrompt(teamsfx, "TeamsBotSsoPrompt", {
+        scopes: ["User.Read"],
+      })
+    );
+    ```
+
+    2. Begin the dialog and sign-in.
+
+    ```ts
+    dialogs.add(
+      new WaterfallDialog("taskNeedingLogin", [
+        async (step) => {
+          return await step.beginDialog("TeamsBotSsoPrompt");
+        },
+        async (step) => {
+          const token = step.result;
+          if (token) {
+            // ... continue with task needing access token ...
+          } else {
+            await step.context.sendActivity(`Sorry... We couldn't log you in. Try again later.`);
+            return await step.endDialog();
+          }
+        },
+      ])
+    );
+    ```
+
+    For more information on sample to use graph API in bot application, see the [bot-sso sample](https://github.com/OfficeDev/TeamsFx-Samples/tree/dev/bot-sso).
+
+    </details>
+
+    <details>
+    <summary><b>Call Azure Function in tab app: On-Behalf-Of flow</b></summary>
+
+    1. Use `axios` library to make HTTP request to Azure Function.
```

Comment:
The code below is not use `axios` library.


You can follow the code and description below as reference to organize the doc:

Use `axios` library to call Azure Function:
```
const accessToken = await teamsfx.getCredential().getToken(""); // Get SSO token 
// teamsfx.getConfig("apiEndpoint") will read REACT_APP_FUNC_ENDPOINT environment variable 
const endpoint = teamsfx.getConfig("apiEndpoint");
const response = await axios.default.get(endpoint + "/api/" + functionName, {
  headers: {
    authorization: "Bearer " + accessToken.token,
  },
});
 ```

You can also use CreateApiClient  provided by teamsfx sdk to call Azure Function:
```
const teamsfx = new TeamsFx();
const credential = teamsfx.getCredential(); //Create an API Client that uses SSO token to authenticate requests
const apiClient = CreateApiClient(teamsfx.getConfig("apiEndpoint")),
new BearerTokenAuthProvider(async () =>  (await credential.getToken(""))!.token);
const response = await apiClient.get("/api/" + functionName); 
```

Reply:
Fine. I'll consider it.